consensus/ibft/pos.go

```diff
@@ -110,12 +110,12 @@ func (pos *PoSMechanism) acceptStateLogHook(snapParam interface{}) error {
 // insertBlockHook checks if the block is the last block of the epoch,
 // in order to update the validator set
 func (pos *PoSMechanism) insertBlockHook(numberParam interface{}) error {
-	headerNumber, ok := numberParam.(uint64)
+	block, ok := numberParam.(*types.Block)
 	if !ok {
 		return ErrInvalidHookParam
 	}
 
-	return pos.updateValidators(headerNumber)
+	return pos.updateValidators(block.Number())
```

Comment:
Why change this to receive a block if you aren't going to use other fields in the block (and PoA doesn't implement this)?